File: greedy/1509_minimum_difference_between_largest_and_smallest_value_in_three_moves.ts
function minDifference(nums: number[]): number {

    let ans = Infinity;
    let poss = [[0, -4], [1, -3], [2, -2], [3, -1]];

    if(nums.length <= 3) {
        return 0;
    }

    nums.sort((a,b) => a-b);

    for(let p of poss){
        let start = p[0];
        let end =  p[1];

        let curr = nums[end + nums.length] - nums[start];
        ans = Math.min(curr, ans);

    }

    return ans

};

File: greedy/1509_minimum_difference_between_largest_and_smallest_value_in_three_moves.rs
impl Solution {
    pub fn min_difference(mut nums: Vec<i32>) -> i32 {

        let poss = [[0, -4], [1, -3], [2, -2], [3, -1]];
        let mut ans:i32 = i32::MAX;
        nums.sort();

        if nums.len() <= 3 {
            return 0;
        }

        for [start, end] in poss.iter() {
            let curr = (nums[*end as usize + nums.len()] - nums[*start as usize]).abs();
            ans = i32::min(ans, curr);
        }

        ans

    }
}